shacl12-vocabularies/shnex-sparql.ttl

```diff
@@ -0,0 +1,957 @@
+## --------
+
+# W3C SHACL Node Expressions Vocabulary
+
+# THIS VERSION IS UNDER DEVELOPMENT BY THE DATA-SHAPES (SHACL 1.2) WG
+
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+@prefix rdf:   <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix xsd:   <http://www.w3.org/2001/XMLSchema#> .
+@prefix sh:    <http://www.w3.org/ns/shacl#> .
+@prefix shnex: <http://www.w3.org/ns/shnex#> .
+@prefix shnex-sparql: <http://www.w3.org/ns/shnex-sparql#> .
+@prefix sparql:  <http://www.w3.org/ns/sparql#> .
+
+shnex-sparql:
+	a owl:Ontology ;
+	rdfs:label "W3C SHACL SPARQL Functions Vocabulary"@en ;
+	rdfs:comment "This vocabulary defines SPARQL functions for use with SHACL."@en ;
+	sh:declare [
+		sh:prefix "sparql" ;
+		sh:namespace "http://www.w3.org/ns/sparql#"^^xsd:anyURI ;
+	] ;
+	owl:imports <http://www.w3.org/ns/shnex#>, <http://www.w3.org/ns/sparql-query#>  .
+
+
+shnex-sparql:category
+	a rdf:Property ;
+	rdfs:label "category"@en ;
+	rdfs:comment "Indicates the category of a SPARQL function, such as algebraic, numeric, string, date, aggregate, or other."@en ;
+	rdfs:domain [
+		a owl:Class ;
+		owl:unionOf (
+			sparql:Function
+			sparql:Aggregate
+			sparql:FunctionalForm
+		)
+	] ;
+	rdfs:range xsd:string ;
+	.
+
+shnex-sparql:example
+	a rdf:Property ;
+	rdfs:label "example"@en ;
+	rdfs:comment "Provides an example usage of the SPARQL function or form."@en ;
+	rdfs:domain [
+		a owl:Class ;
+		owl:unionOf (
+			sparql:Function
+			sparql:Aggregate
+			sparql:FunctionalForm
+		)
+	] ;
+	rdfs:range xsd:string ;
+	.
+
+shnex-sparql:infixOperator
+	a rdf:Property ;
+	rdfs:label "infix operator"@en ;
+	rdfs:comment "Indicates the infix operator symbol used in SPARQL syntax for this function, enabling conversion from functional form to infix notation."@en ;
+	rdfs:domain sparql:Function ;
+	rdfs:range xsd:string ;
+	.
+
+
+# Spec link pattern:
+# For functions in the source whose rdfs:isDefinedBy is the generic spec homepage
+# <http://www.w3.org/TR/sparql12-query/> a per-function anchor was constructed
+# as <http://www.w3.org/TR/sparql12-query/#func-LOCALNAME> where LOCALNAME is
+# the SPARQL local name (e.g., sparql:str -> #func-str). Where a function had a
+# more specific anchor in the source, the same target was used. (This comment
+# documents the fragment pattern used for constructed links.)
+
+# Imported SPARQL function terms (rdf:type, rdfs:label, rdfs:comment, shnex-sparql:category, rdfs:isDefinedBy)
+# Category values: algebraic, numeric, string, date, aggregate, other
+
+
+
+
+
+sparql:plus
+	a sparql:Function ;
+	rdfs:label "plus"@en ;
+	rdfs:comment "This operator adds two numeric expressions and returns their sum."@en ;
+	shnex-sparql:category "algebraic" ;
+	shnex-sparql:infixOperator "+" ;
+	rdfs:isDefinedBy <http://www.w3.org/TR/sparql12-query/> ;
+	shnex-sparql:example """[ sparql:plus ( 38 4 ) ]""" ;
+	.
+
+sparql:subtract
```

Comment:
This has the same URI as the entry in `/ns/sparql.ttl`.

But both graphs have triples with subject `sparql:subtract` and properties `rdfs:comment` and `rdfs:isDefinedBy` with different object.

It would be better to only include the SHACL Node Expression additions.

Reply:
I guess Andy's comment was about the subjects such as sparql:subtract. Both documents use rdfs:comment and rdfs:isDefinedBy. But IMHO those values should align. So if we in the SHACL WG find incorrect comments, the same fix should be applied to the /ns/sparql.ttl

Having said this, I don't think we need to owl:import the /ns/sparql.ttl graph so it will unlikely affect anyone.

In the interest of moving forward, I'll mark this as resolved but please reopen if anyone disagrees.